packages/syft/src/syft/core/node/new/task/task_service.py

```diff
@@ -0,0 +1,400 @@
+# stdlib
+from datetime import date
+from io import StringIO
+import os
+import sys
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import cast
+
+# third party
+import requests
+from result import Err
+from result import Ok
+from result import Result
+
+# relative
+from .....logger import logger
+from .....oblv.constants import LOCAL_MODE
+from ....common.serde.deserialize import _deserialize as deserialize
+from ....common.serde.serializable import serializable
+from ....common.uid import UID
+from ...common.node_table.syft_object import SYFT_OBJECT_VERSION_1
+from ...common.node_table.syft_object import SyftObject
+from ..action_object import ActionObject
+from ..action_store import ActionStore
+from ..api import SyftAPI
+from ..context import AuthedServiceContext
+from ..document_store import DocumentStore
+from ..service import AbstractService
+from ..service import service_method
+from .task import NodeView
+from .task import Task
+from .task_stash import TaskStash
+
+DOMAIN_CONNECTION_PORT = int(os.getenv("DOMAIN_CONNECTION_PORT", 3030))
+
+
+@serializable(recursive_serde=True)
+class DictObject(SyftObject):
+    # version
+    __canonical_name__ = "Dict"
+    __version__ = SYFT_OBJECT_VERSION_1
+
+    base_dict: Dict[Any, Any] = {}
+
+    # serde / storage rules
+    __attr_state__ = ["id", "base_dict"]
+
+    __attr_searchable__ = []
+    __attr_unique__ = ["id"]
+
+
+stdout_ = sys.stdout
+stderr_ = sys.stderr
+
+
+@serializable(recursive_serde=True)
+class TaskService(AbstractService):
+    document_store: DocumentStore
+    task_stash: TaskStash
+    action_store: ActionStore
+
+    def __init__(
+        self, document_store: DocumentStore, action_store: ActionStore
+    ) -> None:
+        self.document_store = document_store
+        self.task_stash = TaskStash(store=document_store)
+        self.action_store = action_store
+
+    @service_method(path="task.create_enclave_task", name="create_enclave_task")
+    def create_enclave_task(
+        self,
+        context: AuthedServiceContext,
+        inputs: Dict[NodeView, dict],
+        code: str,
+        outputs: List[str],
+        owners: List[NodeView],
+        task_id: UID,
+        oblv_metadata: Optional[dict] = None,
+    ) -> Result[Ok, Err]:
+        """Enclave Submit task"""
+        # TODO 🟣 Check for permission after it is fully integrated
+
+        if owners and not isinstance(owners, list):
```

Comment:
Q: In what case we might not have a list of owners

Reply:
Good catch, with the new services architecture we would do the type checking before sending a request on the client side itself, I mainly wanted to sanity check that they are passing at least one owner when doing computation on an enclaves
I have modified it in [b1c9af5](https://github.com/OpenMined/PySyft/pull/7289/commits/b1c9af5f30c99117d59aa92dc15e8c9698d49b96)